src/server/game/Warden/Warden.h

```diff
@@ -113,6 +113,10 @@ class Warden
     virtual void InitializeModule() = 0;
     virtual void RequestHash() = 0;
     virtual void HandleHashResult(ByteBuffer &buff) = 0;
+    virtual uint16 RegisterPayload(std::string& payload) = 0;
```

Comment:
Are you planning on modifying the payload string? If not pass it by const reference.

Reply:
Fixed, thank you. :)